pynamodb/attributes.pyi

```diff
@@ -140,20 +111,16 @@ class MapAttribute(Generic[_KT, _VT], Attribute[Mapping[_KT, _VT]], metaclass=Ma
     def __getattr__(self, attr: str) -> _VT: ...
     def __getitem__(self, item: _KT) -> _VT: ...
     def __set__(self, instance: Any, value: Union[None, MapAttribute[_KT, _VT], Mapping[_KT, _VT]]) -> None: ...
-    @overload
+    @overload  # type: ignore
     def __get__(self: _A, instance: None, owner: Any) -> _A: ...
     @overload
-    def __get__(self: _MT, instance: Any, owner: Any) -> _MT: ...
+    def __get__(self: _MT, instance: _AttributeContainer, owner: Any) -> _MT: ...
     def is_type_safe(self, key: Any, value: Any) -> bool: ...
     def validate(self) -> bool: ...
 
-class ListAttribute(Generic[_T], Attribute[List[_T]]):
+class ListAttribute(Generic[_MT], Attribute[List[_MT]]):
```

Comment:
oops, that's true only if `of is not None`, otherwise every element of ListAttribute can be either a number, a string, a MapAttribute, etc. but we cannot really express it in type stubs alone :(